common/lib/capa/capa/responsetypes.py

```diff
@@ -1097,6 +1097,9 @@ def get_score(self, student_answers):
                 cmap.set(aid, 'correct')
             else:
                 cmap.set(aid, 'incorrect')
+            answer_variable = self.get_student_answer_variable_name(student_answers, aid)
```

Comment:
@symbolist I think there is no need to override ```evaluate_answers()``` because we are creating ```correct_map``` in ```get_score()```, we need to add ```answervariable``` in ```correct_map``` object and we are getting student answers in ```get_score()``` and looping on them to get answer is ```correct``` or ```incorrect```,  so this is the place where we need to get student answer variable name for every answer.

If you want to do this ```answer_variable = self.get_student_answer_variable_name(student_answers, aid)``` in ```evaluate_answers()``` then we have to call ```self.get_answers()``` again in it and again loop on every answer, i don't think so this is a good approach to do loop on answers in two different places because we are doing it in ```get_score()``` already.